### DISCUSSION TITLE
Does Anonymous Sign-In create new users every time?

### DISCUSSION THREAD

**Discussion by azalessky** [supabase/supabase#23383] [Questions] [answered]
I have a Flutter app which allows users to sign-in anonymously and operate without providing their social logins.

I noticed that Supabase creates a new anonymous user in the Auth table after each start of the app.

Is there a way to keep the existing UID created first time in the app like Firebase does?

**Comment by dshukertjr** [comment, marked as answer]
@npopok Calling `signInAnonumously()` will create a new user every time you call it. What you should do is to check if the user is already signed in or not, and call `signInAnonymously()` only when the user is not signed in.

**Reply by azalessky** [reply]
Thanks for the reply @dshukertjr 
I figured it out after some debugging. It was a bit confusing as it works differently comparing to Firebase.
Anyway, I came up with the following check:
  if (Supabase.instance.client.auth.currentUser == null) {
    await Supabase.instance.client.auth.signInAnonymously();
  }
It seems to be working well for me.

**Reply by dshukertjr** [reply]
Yup, that's the way to go.

**Reply by harrisrobin** [reply]
thanks for this, this was useful. I do agree that it is confusing and i would have expected the SDK to handle this for me, similar to firebase.

Thanks 👍🏽

**Reply by mo-tech55** [reply]
I do the same check but with currentSession. I think sessions in supabase live really long but dont really know what the difference. I found this in the docs:

```
  /// Returns the current logged in user, if any;
  ///
  /// Use [currentSession] to determine whether the user has an active session,
  /// because [currentUser] can be non-null without an active session, such as
  /// when the user signed up using email and password but has not confirmed
  /// their email address.
  User? get currentUser => _currentUser;

  /// Returns the current session, if any;
  Session? get currentSession => _currentSession;
```

and this is my flutter check
```
final session = _supabaseClient.auth.currentSession;
    if (session == null) {
      print('No active session found - initiating anonymous sign in');
      add(SignInAnonymously());
    } else {
```

**Comment by mo-tech55** [comment]
I do the same check but with currentSession. I think sessions in supabase live really long but dont really know what the difference. I found this in the docs:

  ```
/// Returns the current logged in user, if any;
  ///
  /// Use [currentSession] to determine whether the user has an active session,
  /// because [currentUser] can be non-null without an active session, such as
  /// when the user signed up using email and password but has not confirmed
  /// their email address.
  User? get currentUser => _currentUser;

  /// Returns the current session, if any;
  Session? get currentSession => _currentSession;

```
and this is my flutter check

```
final session = _supabaseClient.auth.currentSession;
    if (session == null) {
      print('No active session found - initiating anonymous sign in');
      add(SignInAnonymously());
    } else {
```
